src/metadata.ts

```diff
@@ -32,20 +32,27 @@ type WithMeta<T extends z.ZodTypeAny> = MetaFixForStrippedObject<T> & {
   example: ExampleSetter<T>;
 };
 
-export const withMeta = <T extends z.ZodTypeAny>(schema: T) => {
-  const def = schema._def as MetaDef<T>;
+const cloneSchemaForMeta = <T extends z.ZodTypeAny>(schema: T): WithMeta<T> => {
+  const This = (schema as any).constructor;
+  const def = clone(schema._def) as MetaDef<T>;
   def[metaProp] = def[metaProp] || { examples: [] };
-  if (!("example" in schema)) {
-    Object.defineProperties(schema, {
-      example: {
-        get: (): ExampleSetter<T> => (value) => {
-          def[metaProp].examples.push(value);
-          return schema as WithMeta<T>;
-        },
+  return new This({ ...def }) as WithMeta<T>;
+};
+
+export const withMeta = <T extends z.ZodTypeAny>(schema: T): WithMeta<T> => {
+  const result = cloneSchemaForMeta<T>(schema);
```

Comment:
WithMeta on WithMeta makes redundant nesting that requires to be explicit.
Can I make it more smart?